pkg/controller/deployment/custom_overrides_test.go

```diff
@@ -0,0 +1,56 @@
+package deployment
+
+import (
+	"testing"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/stretchr/testify/require"
+
+	"github.com/openshift/cert-manager-operator/apis/operator/v1alpha1"
+	"github.com/openshift/cert-manager-operator/pkg/operator/assets"
+)
+
+func TestCustomOverrides(t *testing.T) {
+	deploymentFile, err := assets.Asset("cert-manager-deployment/controller/cert-manager-deployment.yaml")
+	require.NoError(t, err)
+	testDeployment := resourceread.ReadDeploymentV1OrDie(deploymentFile)
+	defaultArgs := testDeployment.Spec.Template.Spec.Containers[0].Args
+
+	type TestData struct {
+		overrides *v1alpha1.UnsupportedConfigOverrides
+		wantArgs  []string
+	}
+	tests := map[string]TestData{
+		"Empty config overrides should not touch the deployment": {
+			overrides: &v1alpha1.UnsupportedConfigOverrides{},
+			wantArgs:  defaultArgs,
+		},
+		"Other config overrides should not touch the controller": {
+			overrides: &v1alpha1.UnsupportedConfigOverrides{
+				Webhook: v1alpha1.UnsupportedConfigOverridesForCertManagerWebhook{
+					Args: []string{"test"},
+				},
+				CAInjector: v1alpha1.UnsupportedConfigOverridesForCertManagerCAInjector{
+					Args: []string{"test"},
+				},
+			},
+			wantArgs: defaultArgs,
+		},
+		"Controller overrides should work": {
+			overrides: &v1alpha1.UnsupportedConfigOverrides{
+				Controller: v1alpha1.UnsupportedConfigOverridesForCertManagerController{
+					Args: []string{"test"},
+				},
+			},
+			wantArgs: append(defaultArgs, "test"),
```

Comment:
I just noticed, this is absolutely wrong!!!

We should be overriding settings not appending new ones.